### ISSUE TITLE
Readme Low level messaging: bus connection example error

### ISSUE THREAD

**Issue by JetForMe** [sidorares/node-dbus#94]
The docs talk about "dbus.createClient(options)", but the example shows "dbus.createConnection()."

**Comment by sidorares** [comment]
oops. Do you know which one is correct ( don't have time to read through code right now )

**Comment by sidorares** [comment]
Any improvements to documentations are very much welcome, if you find time adding something please submit PR

**Comment by JetForMe** [comment]
Yeah, I don't know which is correct! I'm trying some experiments now. I'm having some issues getting DBus to play nicely in all of a handful of configurations, so I was trying some changes. I think `createConnection()` is right.

**Comment by sidorares** [comment]
they both are correct, that was probably not a good choice of names

https://github.com/sidorares/node-dbus/blob/master/index.js#L129-L132

'connection' is for low level messaging - `conn.message({...})` and `conn.on('message, ...)'`

'client' handles RPC-style calls by remembering association from serial/replySerial to callback

**Comment by sidorares** [comment]
and you can access connection from a client:

```js
var bus = require('dbus-native').systemBus(); // "client"
bus.connection.message({ body: 'hello', signature: 's'}); 
```

**Comment by JetForMe** [comment]
Heh. I find this all very confusing. I'm using DBus to communicate between a C++ app and my node.js app, on both OS X and Debian. During development, I want to be able to launch one or the other, or both, in any order, and I want them to quickly connect or abandon a connection attempt and connect later. I wish I could do this without dbus-daemon running at all, but that doesn't seem to be the case.

**Comment by sidorares** [comment]
this module allows to communicate directly - you could pretend that you are daemon and point C++ app to your node app by setting DBUS_SESSION_BUS_ADDRESS

**Comment by JetForMe** [comment]
But it seems I have to create a socket somewhere.

**Comment by sidorares** [comment]
Both are correct — they return different things, and the README no longer implies otherwise.

| | returns | for |
| --- | --- | --- |
| `dbus.createConnection(opts)` | `DBusConnection` | the transport alone — `conn.message({…})`, `conn.on('message', …)` |
| `dbus.createClient(opts)` | `MessageBus` | RPC on top of it: serial↔callback tracking, `invoke`, proxies, signals |

`sessionBus()` and `systemBus()` are `createClient` with a `busAddress` filled in, and a client always exposes its connection as `bus.connection`, so you can drop to the lower level at any point:

```js
const bus = require('dbus-native').systemBus();
bus.connection.message({ /* … */ });
```

Both are now documented with that distinction in [docs/api.md#entry-points](https://github.com/sidorares/dbus-native/blob/master/docs/api.md#entry-points), which did not exist when this was filed.

On @JetForMe's follow-up about talking directly between two apps with no `dbus-daemon`: that does work, and is what the `direct: true` option is for — it skips the opening `Hello`, which only a bus daemon answers. One side listens (`dbus.createServer`) and the other connects to it. Be aware `lib/server-handshake.js` is still a stub that replies with a hardcoded GUID, so treat the server side as experimental; it is flagged as such in the docs.
